### ISSUE TITLE
Missing required argument doesn't cause an error when that argument type is boolean

### ISSUE THREAD

**Issue by CarlosCuevas** [strongloop/loopback#2092]
```javascript
Person.remoteMethod('createPerson', {
        accepts: [
            { arg: 'name', type: 'string', required: true },
            { arg: 'age', type: 'number', required: true },
            { arg: 'is_alive', type: 'boolean', required: true }
        ],
        http: {
            path: '/',
            verb: 'post'
        }
    });
```
If I send a request to this endpoint that is missing 'name' or 'age', it will return an error.  If I send a request missing 'is_alive', it will proceed with 'is_alive' set to false.

**Comment by Neil-UWA** [comment]
I guess that you  may have set a default value for is_alive in your model defination

**Comment by CarlosCuevas** [comment]
@Neil-UWA Thanks, but there are many cases where it would be inappropriate to set a default value (exactly when you'd want to set a field as required).

**Comment by richardpringle** [comment]
Hey @CarlosCuevas, I'm guessing that somewhere in the code that `is_alive = undefined` is evaluating to `false`. I will look into the details of the issue.

Thanks!

**Comment by richardpringle** [comment]
@bajtos, I found the source of the issue: https://github.com/strongloop/strong-remoting/blob/master/lib/dynamic.js#L98-L117

```node
Boolean(undefined) // returns false;
```
This is an easy fix, but before I do so, I just want to double check to make sure there aren't any use-cases where you wouldn't want the argument to remain "undefined".

**Comment by bajtos** [comment]
@richardpringle thanks for checking, a change like this can have subtle side effects, so it's best to consider it thoroughly.

We have recently discovered quite few flaws in strong-remoting's argument coercion (see strongloop/strong-remoting#264, strongloop/loopback#1806, strongloop/strong-remoting#208), there is a spike story to investigate the issues and come up with a more robust design, see https://github.com/strongloop-internal/scrum-loopback/issues/706. 

So much for the background context.

As for fixing boolean conversion, see also https://github.com/strongloop/strong-remoting/commit/b924a79f23a97eaa294bbd60345fb0f0cc77c5a0 (which has been reverted as it did not work well).

>  I found the source of the issue: https://github.com/strongloop/strong-remoting/blob/master/lib/dynamic.js#L98-L117
> ```js
return Boolean(undefined) // returns false;
// we should return undefined instead
```
> This is an easy fix, but before I do so, I just want to double check to make sure there aren't any use-cases where you wouldn't want the return value to remain "undefined"

I am inclining towards changing boolean converter to preserve undefined values, as you are proposing @richardpringle.

@ritch @raymondfeng @STRML do you have any opinion on this? Can you see a use case where the proposed change would be a problem?

**Comment by STRML** [comment]
FYI @bajtos we can't see those internal issues.

**Comment by bajtos** [comment]
> @STRML we can't see those internal issues.

That internal issue is just a placeholder that helps us to plan our work in sprints. It lists the three issues I mentioned in my comment above - strongloop/strong-remoting#264, strongloop/loopback#1806, strongloop/strong-remoting#208.

I know the internal issues are not ideal, but let's keep that matter out of the discussion in this particular GH issue.

**Comment by STRML** [comment]
Not familiar with the term "spike story", assumed there was something of
substance since you linked it.

**Comment by richardpringle** [comment]
@STRML: Spike - research and come up with a proposal for a change.

There's nothing of substance in "spike" story, @bajtos just wanted to let me know that we are aware there is a problem.

**Comment by CarlosCuevas** [comment]
@richardpringle Just caught that this is also the case when type is "array".

```js
Person.remoteMethod('createPerson', {
        accepts: [
            { arg: 'name', type: 'string', required: true },
            { arg: 'age', type: 'number', required: true },
            { arg: 'children', type: 'array', required: true }
        ],
        http: {
            path: '/',
            verb: 'post'
        }
    });
```

failing to passing children will not trigger an error

**Comment by bajtos** [comment]
Related: #603

> I expected the method to return a 400 bad request or something when the argument payload (i.e.: the body of the post in the case) isn't an array of objects. However, it seems to take any valid json.

**Comment by bajtos** [comment]
The problem will be fixed in LoopBack 3.0, see https://github.com/strongloop/strong-remoting/pull/343.
